### PR TITLE
refactor: add slippage to orders

### PR DESCRIPTION
**Pull request by witgaw**
- include open orders in slippage calculation
- formalise expected behaviour on new order (initial level required unless party tries to reduce their position).

### CODE REVIEW

File: protocol/0019-MCAL-margin_calculator.md [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 - Zero position and zero orders results in all zero margin levels (<a name="0019-MCAL-003" href="#0019-MCAL-003">0019-MCAL-003</a>)
 
-- If `riskiest long > 0` and there are no bids on the order book, the `exit price` is equal to the initial mark price, as set by a market parameter. (<a name="0019-MCAL-004" href="#0019-MCAL-004">0019-MCAL-004</a>)
```
Comment (Jiajia-Cui) [inline protocol/0019-MCAL-margin_calculator.md:11]: If `riskiest long > 0 && 0 < *sum of volume of order book bids* < riskiest long`, the `exit price` is equal to infinity

Comment (witgaw) [reply protocol/0019-MCAL-margin_calculator.md:11]: Good spot, thanks. Corrected now and added more (incremented code numbers for each).

**Review by Jiajia-Cui — DISMISSED** [review]
LGTM

**Review by gordsport — APPROVED** [review]
reapproving as last commit was just to fix an MD lint error (no content change)

**Pull request merged**